platform/security/handler/security-handler-saml/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -204,7 +204,8 @@ public Response sendLogoutRequest(@QueryParam("EncryptedNameIdTime") String encr
 
         logout();
         if (logoutMessage == null) {
-          throw new RuntimeException("Logout message not available yet.");
+          LOGGER.info("Logout message not available yet");
+          return buildLogoutResponse(UNABLE_TO_CREATE_LOGOUT_REQUEST);
```

Comment:
❓ The two different `logoutMessage == null` checks return different responses. Why an error for the other and not here?

Reply:
I wasn't sure what to do here. In the `sendLogoutRequest` method, `buildLogoutResponse("some message")` is returned in error cases. In the `soapLogoutRequest` method, `Response.serverError().build()` is returned in error cases.

What do you think would be the best to return in these 2 places?

Reply:
I had to look how those endpoints are used again. `sendLogoutRequest` is the logout action endpoint where we redirect users for a SAML logout, so it makes sense to return a redirect. `soapLogoutRequest` is the SLO endpoint which is where the SAML IdP sends LogoutRequests, and returning a redirect to the IdP doesn't make sense.

All that to say I think it's fine like you have it 😄